File: codes/php/array_includes.php
<?php

echo("\n// JavaScript-like Array.includes() in PHP" . "\n");
    
$my_friends = ['Alisa', 'Trivia'];
echo("my friends: " . str_replace(",", ", ", str_replace(":", ": ", json_encode($my_friends))) . "\n");

$a_name = 'Alisa';
$is_my_friend = in_array($a_name, $my_friends);
echo("is my friends includes \"$a_name\": " . json_encode($is_my_friend) . "\n");
// is my friends includes "Alisa": true

$a_name = 'Trivia';
$is_my_friend = in_array($a_name, $my_friends);
echo("is my friends includes \"$a_name\": " . json_encode($is_my_friend) . "\n");
// is my friends includes "Trivia": true

$a_name = 'Tony';
$is_my_friend = in_array($a_name, $my_friends);
echo("is my friends includes \"$a_name\": " . json_encode($is_my_friend) . "\n");
// is my friends includes "Tony": false

$a_name = 'Ezekiel';
$is_my_friend = in_array($a_name, $my_friends);
echo("is my friends includes \"$a_name\": " . json_encode($is_my_friend) . "\n");
// is my friends includes "Ezekiel": false

File: codes/csharp/array_includes.cs
using System; // Console, Func<>, Action<>, Action
using System.Collections.Generic; // Dictionary<>, List<>, KeyValuePair<>, IEnumerable<>

class Program {
    static void Main(string[] Args) {
        dynamic IsNumeric = (Func<dynamic, bool>)((dynamic Anything) => (Anything is sbyte || Anything is byte || Anything is short || Anything is ushort || Anything is int || Anything is uint || Anything is long || Anything is ulong || Anything is float || Anything is double || Anything is decimal));

        string PrettyArrayOfPrimitives(List<dynamic> AnArrayOfPrimitives) {
            string Result = "[";
            int ArrayItemIndex = 0;
            foreach (dynamic ArrayItem in AnArrayOfPrimitives) {
                if (((ArrayItem is string) == false) && (IsNumeric(ArrayItem) == false) && ((ArrayItem is bool) == false) && (ArrayItem != null)) continue;
                if (ArrayItem is string) Result += $"\"{ArrayItem}\"";
                if (IsNumeric(ArrayItem)) Result += ArrayItem.ToString().Replace(",", ".");
                if (ArrayItem is bool) Result += $"{ArrayItem.ToLower()}";
                if (ArrayItem == null) Result += "null";
                if ((ArrayItemIndex + 1) != AnArrayOfPrimitives.Count) Result += ", ";
                ArrayItemIndex += 1;
            }
            Result += "]";
            return Result;
        }

        Console.WriteLine("\n// JavaScript-like Array.includes() in C#");

        dynamic MyFriends = new List<dynamic>() {"Alisa", "Trivia"};
        Console.WriteLine($"MyFriends: {PrettyArrayOfPrimitives(MyFriends)}");

        dynamic AName;
        dynamic IsMyFriend;

        AName = "Alisa";
        IsMyFriend = MyFriends.Contains(AName);
        Console.WriteLine($"is my friends includes \"{AName}\": {IsMyFriend}");
        // is my friends includes "Alisa": True

        AName = "Trivia";
        IsMyFriend = MyFriends.Contains(AName);
        Console.WriteLine($"is my friends includes \"{AName}\": {IsMyFriend}");
        // is my friends includes "Trivia": True

        AName = "Tony";
        IsMyFriend = MyFriends.Contains(AName);
        Console.WriteLine($"is my friends includes \"{AName}\": {IsMyFriend}");
        // is my friends includes "Tony": False

        AName = "Ezekiel";
        IsMyFriend = MyFriends.Contains(AName);
        Console.WriteLine($"is my friends includes \"{AName}\": {IsMyFriend}");
        // is my friends includes "Ezekiel": False

    }
}
